### ISSUE TITLE
Incorrect Boolean Cast in BoxLib frontend

### ISSUE THREAD

**Issue by dwillcox** [yt-project/yt#1618]
### Bug report

**Bug summary**

The BoxLib frontend incorrectly interprets boolean dataset parameters read from the `job_info` file for Maestro datasets.

These boolean parameters are written as `"T"` or `"F"` for Maestro datasets. The `_guess_pcast` function correctly determines these are to be cast as `bool` but then does so by calling `bool([parameter])` which will always return `True` since `[parameter]` is a nonzero-length string.

This is not a problem for Castro datasets, where boolean parameters are expressed as integers 0 or 1.

**Code for reproduction**

```python
import yt
# Using the dataset from the yt sample
# data in the yt Data Hub.
ds = yt.load('maestro_subCh_plt00248')
ds.parameters
```

**Actual outcome**

```
...
'use_thermal_diffusion': True,
...
```

**Expected outcome**

```
...
'use_thermal_diffusion': False,
...
```

**Version Information**

  * Operating System: Fedora 25
  * Python Version: Python 3.5.4
  * yt version: git commit c122c97bddf36605f37a2eb122c7fb43c903b464
  * Other Libraries (if applicable): N/A

python installed via Anaconda, yt installed from source (git commit above)

**Comment by dwillcox** [comment]
Fixed via PR #1619

**Comment by ngoldbaum** [comment]
In the future if you say "Fixes #xxxx" in your commit message or PR title, the issue will auto-close.

Hope to hear more from you in the future and thank you for contributing code to yt!
